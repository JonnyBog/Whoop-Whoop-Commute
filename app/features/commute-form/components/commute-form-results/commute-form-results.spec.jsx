import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import CommuteFormResults from 'features/commute-form/components/commute-form-results/commute-form-results';

describe('Features', () => {
  describe('Commute Form Results Component', () => {
    it('should export', () => {
      expect(CommuteFormResults).toBeDefined();
    });

    let props;

    beforeEach(() => {
      props = {
        data: [
          {
            duration: 10,
            legs: [
              {
                departurePoint: {
                  commonName: 'test name'
                },
                instruction: {
                  summary: 'test summary'
                }
              }
            ]
          },
          {
            duration: 1,
            legs: [
              {
                departurePoint: {
                  commonName: 'test name'
                },
                instruction: {
                  summary: 'test summary'
                }
              }
            ]
          }
        ]
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createRenderTree
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} component tree
     */
    function createRenderTree (updatedProps = props) {
      return renderer.create(
        <CommuteFormResults {...updatedProps} />
      ).toJSON();
    }

    /**
     * createShallowWrapper
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} shallow component
     */
    function createShallowWrapper (updatedProps = props) {
      return shallow(
        <CommuteFormResults {...updatedProps} />
      );
    }

    describe('Appearance', () => {
      it('should render', () => {
        const tree = createRenderTree();
        expect(tree).toMatchSnapshot();
      });

      it('should remove any duplicate results with the same Common Name', () => {
        const wrapper = createShallowWrapper();

        expect(wrapper.find('commute-form-results__ResultWrapper')).toHaveLength(1);
      });

      it('should render as many JourneyCopy elements as there are legs', () => {
        const wrapper = createShallowWrapper();

        expect(wrapper.find('commute-form-results__ResultWrapper').at(0).find('commute-form-results__ResultsCopy')).toHaveLength(1);
      });
    });
  });
});
