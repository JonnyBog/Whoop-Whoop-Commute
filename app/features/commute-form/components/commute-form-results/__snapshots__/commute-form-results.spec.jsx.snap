// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Commute Form Results Component Appearance should render 1`] = `
<div
  className="commute-form-results__ResultWrapper-sc-1jiponl-1 hvWEVL"
>
  <h2
    className="commute-form-results__ResultTitle-sc-1jiponl-2 dJVOAy"
  >
    test name
  </h2>
  <p
    className="commute-form-results__ResultsCopy-sc-1jiponl-0 kRINpC"
  >
    10
     
    minutes
  </p>
  <p
    className="commute-form-results__JourneyCopy-sc-1jiponl-3 efvvJE"
  >
    test summary
  </p>
</div>
`;
